pkg/actors/actors.go

```diff
@@ -205,16 +210,37 @@ func (a *actorsRuntime) Call(req *CallRequest) (*CallResponse, error) {
 	if a.isActorLocal(targetActorAddress, a.config.HostAddress, a.config.Port) {
 		resp, err = a.callLocalActor(req.ActorType, req.ActorID, req.Method, req.Data, req.Metadata)
 	} else {
-		resp, err = a.callRemoteActor(targetActorAddress, req.ActorType, req.ActorID, req.Method, req.Data, req.Metadata)
+		resp, err = a.callRemoteActorWithRetry(callRemoteActorRetryCount, a.callRemoteActor, targetActorAddress, daprID, req.ActorType, req.ActorID, req.Method, req.Data, req.Metadata)
 	}
 
 	if err != nil {
 		return nil, err
 	}
-
 	return resp, nil
 }
 
+// callRemoteActorWithRetry will call a remote actor for the specified number of retries and will only retry in the case of transient failures
+func (a *actorsRuntime) callRemoteActorWithRetry(numRetries int, fn func(targetAddress, targetID, actorType, actorID, actorMethod string, data []byte, metadata map[string]string) (*CallResponse, error),
```

Comment:
Can we simply use retry middle ware to call grpc ? 
https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/retry/examples_test.go

Reply:
I'll look into that

Reply:
Just tested it with the middleware, which unfortunately doesn't work and will not recreate the connection in case of a server shutdown.. :(